pandas/tests/io/json/test_pandas.py

```diff
@@ -1032,6 +1032,67 @@ def test_to_jsonl(self):
         assert result == expected
         assert_frame_equal(pd.read_json(result, lines=True), df)
 
+    def test_read_jsonchunks(self):
+        # GH17048: memory usage when lines=True
+        df = pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]})
+
+        def get_strio():
+            return StringIO(df.to_json(lines=True, orient="records"))
+
+        def test_with_chunksize(c):
+            iterator = pd.read_json(get_strio(), lines=True, chunksize=c)
```

Comment:
I don't really like the sub-functions.

Instead use a fixture for ``get_strio`` (and make it a more meaningful name.

Reply:
Happy to rename it. 

I don't think I can use a fixture, though, because the `StringIO` that is returned can only be read once and every comparison needs to read it twice (chunked and unchunked). I'll move it into a separate function for now, but let me know what you think best.